### PR TITLE
[feature] : initialize project structure

### PR DESCRIPTION
**Pull request by brianwachira**
- App setup ( project initialization )

### CODE REVIEW

**Review by makunomark — CHANGES_REQUESTED** [review]
Would it be prudent to consider pinning a specific node/npm version for use when working with this repo? 
Reason: Came across an issue with `package-lock.json`; It was originally generated with `lockfileversion` set to 2, but due to the old version of npm that I am using caused changes on the lockfile. 
Suggestion: to pin on a version that all contributors have to use. Check out [`engines`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines).

**Comment by brianwachira** [comment]
> Would it be prudent to consider pinning a specific node/npm version for use when working with this repo? Reason: Came across an issue with `package-lock.json`; It was originally generated with `lockfileversion` set to 2, but due to the old version of npm that I am using caused changes on the lockfile. Suggestion: to pin on a version that all contributors have to use. Check out [`engines`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines).

Okay, wow I have never known one can pin on a version. Let me look into it.
So ideally I pin on the version I used to generate the project, right?

**Comment by makunomark** [comment]
Yup. The version you used should work 😊

**Comment by brianwachira** [comment]
> Yup. The version you used should work 😊

Done!

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "devDependencies": {
+    "@babel/core": "^7.12.9",
+    "@types/react": "~18.0.14",
+    "@types/react-native": "~0.69.1",
+    "typescript": "~4.3.5"
+  },
+  "private": true,
+  "engines": {
+    "node": "14.7.0"
```
Comment (makunomark) [inline package.json]:
```suggestion
    "node": "14.7.0",
    "npm": "^7.0.0"
```

Apologies for the back and forth, might you also consider adding a version for npm?

Reason: npm v6 and less is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. We can also lock it to a version thats higher than 7.0.0

A sample of the warning:

image

Comment (brianwachira) [reply package.json]: That's okay, lemme do that right away!

**Review by makunomark — APPROVED** [review]
Awesome work @brianwachira !

**Pull request merged**